services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/ExperimentalWaypointMetadata.java

```diff
@@ -0,0 +1,190 @@
+   *
+   * @return a {@link ExperimentalWaypointMetadata.Builder} with the same values set to match
+   *   the ones defined in this {@link ExperimentalWaypointMetadata}
+   */
+  public abstract Builder toBuilder();
+
+  /**
+   * Gson type adapter for parsing Gson to this class.
+   *
+   * @param gson the built {@link Gson} object
+   * @return the type adapter for this class
+   */
+  public static TypeAdapter<ExperimentalWaypointMetadata> typeAdapter(Gson gson) {
+    return new AutoValue_ExperimentalWaypointMetadata.GsonTypeAdapter(gson);
+  }
+
+  /**
+   * Create a new instance of this class by passing in a formatted valid JSON String.
+   *
+   * @param json a formatted valid JSON string defining a Metadata
+   * @return a new instance of this class defined by the values passed inside this static factory
+   *   method
+   */
+  public static ExperimentalWaypointMetadata fromJson(String json) {
+    GsonBuilder gson = new GsonBuilder();
+    gson.registerTypeAdapterFactory(DirectionsAdapterFactory.create());
+    return gson.create().fromJson(json, ExperimentalWaypointMetadata.class);
+  }
+
+  /**
+   * This builder can be used to set the values describing the {@link ExperimentalWaypointMetadata}.
+   */
+  @AutoValue.Builder
+  public abstract static class Builder {
+
+    /**
+     * Object representing experimental value.
+     * <p>
+     * All available experimental values are subject to change at any time.
+     *
+     * @param type type
+     */
+    @NonNull
+    public abstract Builder type(@Nullable String type);
+
+    /**
+     * Object representing experimental value.
+     * <p>
+     * All available experimental values are subject to change at any time.
+     *
+      @param name name
+     */
+    @NonNull
+    public abstract Builder name(@Nullable String name);
+
+    /**
+     * Object representing experimental value.
+     * <p>
+     * All available experimental values are subject to change at any time.
+     *
+      @param chargeTime chargeTime
+     */
+    @NonNull
+    public abstract Builder chargeTime(@Nullable Integer chargeTime);
+
+    /**
+     * Object representing experimental value.
+     * <p>
+     * All available experimental values are subject to change at any time.
+     *
+      @param chargeTo chargeTo
+     */
+    @NonNull
+    public abstract Builder chargeTo(@Nullable Integer chargeTo);
+
+    /**
+     * Object representing experimental value.
+     * <p>
+     * All available experimental values are subject to change at any time.
+     *
+      @param plugType plugType
+     */
+    @NonNull
+    public abstract Builder plugType(@Nullable String plugType);
+
+    /**
+     * Object representing experimental value.
+     * <p>
+     * All available experimental values are subject to change at any time.
+     *
+      @param powerKw powerKw
+     */
+    @NonNull
+    public abstract Builder powerKw(@Nullable Integer powerKw);
```

Comment:
Question regarding capitalization. Watt is usually symbolized with `W` (capital letter) and kilo by `k` which doesn't translate well to snake case :) Is `powerKiloWatt` too verbose?

I'm likely overanalyzing this. Feel free to ignore me :)

Reply:
Yea, this is an unfortunate setup. On top of snake case, code style verification in this project forbids any acronyms in the API signature (more than 2 capital letters next to each other), so `powerKW` cannot be used and I don't think we need to necessarily revisit this now.

I like the `powerKiloWatt` suggestion :+1: